app/scripts/modules/core/src/artifact/expectedArtifact.service.ts

```diff
@@ -0,0 +1,45 @@
+import { copy, module } from 'angular';
+
+import { IArtifact, IExpectedArtifact, IPipeline, IStage } from 'core/domain';
+import { PipelineConfigService } from 'core';
+
+export class ExpectedArtifactService {
+  constructor(private pipelineConfigService: PipelineConfigService) {
+    'ngInject';
+  }
+
+  public summarizeExpectedArtifact(expected: IExpectedArtifact): string {
```

Comment:
Would be more angular-idiomatic to define a custom filter instead of injecting this service into every controller that needs this.

```
function summarizeExpectedArtifact() {
  return function(expected: IExpectedArtifact) {
     (summarizing logic)
  }
}

module(SUMMARIZE_EXPECTED_ARTIFACT_FILTER, [])
  .filter('summarizeExpectedArtifact', summarizeExpectedArtifact);
```

Then in your templates, you can write
```
{{ expected | summarizeExpectedArtifact }}
```

Reply:
:+1: